### PR TITLE
feat(frontend): sidebar component bb-90

### PR DESCRIPTION
**Pull request by ihorLysak**
### Summary
  Implemented sidebar component

https://github.com/user-attachments/assets/8011f1cf-7a9b-4dcd-bcad-bce155f751d1

### Notes
actual implementation of this component has only one button, added second one in the video for demonstration purposes

### CODE REVIEW

File: apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import wheelOFBalance from "~/assets/img/wheel-of-balance.svg";
+import wheelOFBalanceInavtive from "~/assets/img/wheel-of-balance-inactive.svg";
```
Comment (fshabanov) [inline apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts:2]:
```suggestion
import wheelOFBalanceInactive from "~/assets/img/wheel-of-balance-inactive.svg";
```

File: apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+const sidebarLinkIcon = {
```
Comment (fshabanov) [inline apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts:4]: This is a map - mapping a name to icons

File: apps/frontend/src/pages/my-wheel/my-wheel.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Sidebar } from "~/libs/components/components.js";
+
+import styles from "./styles.module.css";
+
+const MyWheel: React.FC = () => {
+	return (
+		<div className={styles["pageContainer"]}>
+			<Sidebar />
+			<div />
```
Comment (fshabanov) [inline apps/frontend/src/pages/my-wheel/my-wheel.tsx:9]: Do we need this empty div?

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const Sidebar: React.FC = () => {
+	const { pathname } = useLocation();
+	return (
+		<div className={styles["sidebarContainer"]}>
+			<div className={styles["logoContainer"]}>Logo</div>
+			<div className={styles["buttonsContainer"]}>
+				<SidebarLink
+					icon={sidebarLinkIcon.MY_WHEEL}
+					pathname={pathname}
+					title="My Wheel"
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:17]: Create enum for that

File: apps/frontend/src/index.tsx [hunk 5]
```diff
@@ -14,6 +14,8 @@ import { Auth } from "~/pages/auth/auth.jsx";
 import { NotFound } from "~/pages/not-found/not-found.jsx";
 import { Quiz } from "~/pages/quiz/quiz.jsx";
 
+import { MyWheel } from "./pages/my-wheel/my-wheel.js";
```
Comment (fshabanov) [inline apps/frontend/src/index.tsx:17]:
```suggestion
import { MyWheel } from "~/pages/my-wheel/my-wheel.js";
```
To have consistency with earlier imports

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/my-wheel/my-wheel.tsx (hunk 3 above)
Comment (ihorLysak) [reply apps/frontend/src/pages/my-wheel/my-wheel.tsx:9]: Not really, it is just for demonstration purposes. The rest of the page should be in place of this div

File: apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts (hunk 1 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts:2]: What this change introduces?

Comment (fshabanov) [reply apps/frontend/src/libs/enums/sidebar-link-icon.enum.ts:2]: There was a type in the word "Inactive"

File: apps/frontend/src/pages/my-wheel/my-wheel.tsx (hunk 3 above)
Comment (fshabanov) [reply apps/frontend/src/pages/my-wheel/my-wheel.tsx:9]: You can remove for now

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx (hunk 4 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/sidebar.tsx:17]: You mean moving title to already existing enum? Im actually al little confused whether storing icon/button details as an enums satisfies the quality criteria, because as I understand it you suggested remaking this enum into a map in the previous comment

Comment (fshabanov) [reply apps/frontend/src/libs/components/sidebar/sidebar.tsx:17]: I was talking about moving title to enum, but actually let's create an array with sidebar items. Each item should have title, href and icon

Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/sidebar.tsx:17]: where should I store this array? In the same file?

File: apps/frontend/src/libs/components/sidebar/utils/sidebar-items.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import wheelOfBalance from "~/assets/img/wheel-of-balance.svg";
+import wheelOfBalanceInactive from "~/assets/img/wheel-of-balance-inactive.svg";
+import { AppRoute } from "~/libs/enums/enums.js";
+import { SidebarItem } from "~/libs/types/sidebar-item.type.js";
+
+const sidebarItems: SidebarItem[] = [
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/utils/sidebar-items.ts:6]: Should be in `constants` folder

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/sidebar/components/sidebar-link/styles.module.css [hunk 7]
```diff
@@ -0,0 +1,4 @@
+.linkPicture {
```
Comment (ViktoriiaKozopas) [inline apps/frontend/src/libs/components/sidebar/components/sidebar-link/styles.module.css:1]: Let's rename it to `linkIcon` for consistency

File: apps/frontend/src/libs/components/sidebar/styles.module.css [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+	font-size: 28px;
+	font-weight: 800;
+	border-bottom: 1px solid var(--white);
+}
+
+.buttonsContainer {
+	padding: 14px, 12px, 14px, 14px;
+}
+
+.linkPicture {
```
Comment (ViktoriiaKozopas) [inline apps/frontend/src/libs/components/sidebar/styles.module.css:18]: unused duplicate? the same is defined in sidebar-link

File: apps/frontend/src/libs/components/sidebar/utils/sidebar-items.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import wheelOfBalance from "~/assets/img/wheel-of-balance.svg";
+import wheelOfBalanceInactive from "~/assets/img/wheel-of-balance-inactive.svg";
```
Comment (ViktoriiaKozopas) [inline apps/frontend/src/libs/components/sidebar/utils/sidebar-items.ts:2]: I'd probably reverse the names of the files to `wheel-of-balance-active` (note the **active** part) and `wheel-of-balance` for the inactive one - so that the second one will have a _default_ icon, which is inactive in our case. What do you think on that?

**Review by ViktoriiaKozopas — COMMENTED** [review]
Looks good 👍 Left a few comments

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import { useLocation } from "react-router-dom";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:1]: Should be imported from ~/hooks/hooks.js

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+import { sidebarItems } from "~/libs/constants/constants.js";
+
+import { SidebarLink } from "./components/sidebar-link/sidebar-link.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:5]:
```suggestion
import { SidebarLink } from "./libs/components/sidebar-link/sidebar-link.js";
```

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+	const { pathname } = useLocation();
+	return (
+		<div className={styles["sidebarContainer"]}>
+			<div className={styles["logoContainer"]}>Logo</div>
+			<div className={styles["buttonsContainer"]}>
+				{sidebarItems.map((item, index) => {
+					return (
+						<SidebarLink
+							icon={item.icon}
+							key={index}
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:18]: Not a good idea to have index as a key

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import wheelOfBalance from "~/assets/img/wheel-of-balance-active.svg";
+import wheelOfBalanceInactive from "~/assets/img/wheel-of-balance-inactive.svg";
+import { AppRoute } from "~/libs/enums/enums.js";
+import { SidebarItem } from "~/libs/types/sidebar-item.type.js";
+
+const sidebarItems: SidebarItem[] = [
+	{
+		href: AppRoute.ROOT,
+		icon: {
+			active: wheelOfBalance,
```
Comment (fshabanov) [inline apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: Let's create a separate `Icon` component, which will accept icon name
Here you will have icon name

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts (hunk 13 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: I'll use item.title for key then

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx (hunk 12 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/sidebar.tsx:18]: I'll use item.title for key then

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts (hunk 13 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: By here you mean active: IconName in type definition?

Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: type SidebarItem = {
	href: ValueOf<typeof AppRoute>;
	icon: {
		active: IconName;
		inactive: IconName;
	};
	title: string;
};

Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: Screenshot 2024-08-23 at 11 59 54

Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: The reason why I have active and inactive in icon names is because those are intended to be two separate svg files under the hood

Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: Should this Icon component be passed as a child prop to SIdebarLink or used inside SidebarLink's JSX?

Comment (fshabanov) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:10]: The type you sent seems to be correct. To SidebarLink component we will send IconName, which will be used to show the icon

**Comment by ihorLysak** [comment]
### Summary
Current implementation creates a sidebar-icon-source.enum.ts (alternatively this might be a map inside of constant folder).
Keys of this eum are used to create a union type IconName, which is then used as a property on SidebarItem type and SidebarLink component.

File: apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+	title,
+	to,
+}: Properties) => {
+	const isActive = pathname === to;
+	return (
+		<Link isActive={isActive} to={to} type="navLink">
+			<img
+				alt={title}
+				className={styles["linkIcon"]}
+				src={SidebarIconSource[icon]}
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: Create a new Icon component, which will be used like this:
`<Icon iconName="name" />`
No need to have SidebarIconSource

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts [hunk 15]
```diff
@@ -0,0 +1,15 @@
+import { AppRoute } from "~/libs/enums/enums.js";
+import { SidebarItem } from "~/libs/types/sidebar-item.type.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/constants/sidebar-items.constant.ts:2]:
```suggestion
import { SidebarItem } from "~/libs/types/types.js";
```

File: apps/frontend/src/libs/types/icon-name.type.ts [hunk 16]
```diff
@@ -0,0 +1,5 @@
+import { SidebarIconSource } from "~/libs/enums/sidebar-icon-source.enum.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/types/icon-name.type.ts:1]:
```suggestion
import { SidebarIconSource } from "~/libs/enums/enums.js";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx (hunk 14 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: this Icon component will have all of the URIs to svgs inside of it? Or how should it be implemented otherwise?

Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: It still needs to be some sort of a map to retrieve the image url by name, or am I missing something?

Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: Maybe template string?

Comment (fshabanov) [reply apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: We will have a map `iconNameToSvg` from where we will get the required icon. That will be a part of Icon component
All icon svgs will be imported to that map file

Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: Will this map be a separate task or should I implement it right now?

Comment (fshabanov) [reply apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx:26]: You should implement it now - add only the icons that you use for this task

**Comment by ihorLysak** [comment]
Maybe we should pass a style component to icon to regulate it's size?

File: apps/frontend/src/libs/components/icon/icon.tsx [hunk 17]
```diff
@@ -0,0 +1,12 @@
+import { iconNameToSvg } from "~/libs/maps/maps.js";
+import { IconName } from "~/libs/types/icon-name.type.js";
+
+type Properties = {
+	name: IconName;
+};
+
+const Icon: React.FC<Properties> = ({ name }: Properties) => {
```
Comment (yulyaolshanska) [inline apps/frontend/src/libs/components/icon/icon.tsx:8]: I think you should add an alt prop and a prop like "variant" that will be used for the class name

Comment (ihorLysak) [reply apps/frontend/src/libs/components/icon/icon.tsx:8]: I kinda agree on the alt prop, don't know about variant. If I understand you correctly, the usage sould be something like ` variant={styles[sidebarIcon]}`? @fshabanov  what do you think?

Comment (yulyaolshanska) [reply apps/frontend/src/libs/components/icon/icon.tsx:8]: Somethink like in Link component with prop "type"

File: apps/frontend/src/libs/maps/icon-name-to-svg.map.ts [hunk 18]
```diff
@@ -0,0 +1,10 @@
+import wheelOfBalanceActive from "~/assets/img/wheel-of-balance-active.svg";
+import wheelOfBalanceIncative from "~/assets/img/wheel-of-balance-inactive.svg";
+import { IconName } from "~/libs/types/types.js";
+
+const iconNameToSvg = new Map<IconName, string>([
+	["my-wheel-active", wheelOfBalanceActive],
+	["my-wheel-inactive", wheelOfBalanceIncative],
+]);
```
[the remark this answers is not in the document]
Comment (ihorLysak) [reply apps/frontend/src/libs/maps/icon-name-to-svg.map.ts:8]: If we want to store the values as a react element which is an icon, we'll have to install vite-plugin-svgr, and create a separate folder in assets/img for icons, after that configuring the vite-plugin-svgr to target this folder specifically. Otherwise, all svg imports in project will be treated as React.FC<React.SVGProps<SVGSVGElement>>>, which might not be a good behaviour

[the remark this answers is not in the document]
Comment (ihorLysak) [reply apps/frontend/src/libs/maps/icon-name-to-svg.map.ts:8]: Screenshot 2024-08-24 at 15 14 17

[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/frontend/src/libs/maps/icon-name-to-svg.map.ts:8]: Keep `img`, I will create a separate ticket for adding `vite-plugin-svgr`

**Comment by ihorLysak** [comment]
I see an incorrect import, Making a new commit with fixes

File: apps/frontend/packages.d.ts [hunk 19]
```diff
@@ -27,3 +27,10 @@ declare module "eslint-plugin-jsx-a11y" {
 		configs,
 	};
 }
+
+declare module "*.svg" {
+	import React from "react";
+
+	const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>;
+	export default ReactComponent;
+}
```
Comment (fshabanov) [inline apps/frontend/packages.d.ts:36]: What if we add `/// <reference types="vite-plugin-svgr/client" />` to vite-env.d.ts, would that work?

File: apps/frontend/src/libs/components/icon/icon.tsx [hunk 20]
```diff
@@ -0,0 +1,13 @@
+import { iconNameToSvg } from "~/libs/maps/maps.js";
+import { IconName } from "~/libs/types/icon-name.type.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/icon/icon.tsx:2]:
```suggestion
import { IconName } from "~/libs/types/types.js";
```

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+const sidebarItems: SidebarItem[] = [
+	{
+		href: AppRoute.ROOT,
+		icon: {
+			active: "myWheelActive",
+			inactive: "myWheelInactive",
+		},
+		title: "My Wheel",
+	},
+] as const;
```
Comment (fshabanov) [inline apps/frontend/src/libs/constants/sidebar-items.constant.ts:13]:
```suggestion
const SIDEBAR_ITEMS: SidebarItem[] = [
	{
		href: AppRoute.ROOT,
		icon: {
			active: "myWheelActive",
			inactive: "myWheelInactive",
		},
		title: "My Wheel",
	},
];
```

File: apps/frontend/src/libs/maps/icon-name-to-svg.map.ts [hunk 22]
```diff
@@ -0,0 +1,13 @@
+import WheelOfBalanceActive from "~/assets/img/wheel-of-balance-active.svg";
+import WheelOfBalanceInactive from "~/assets/img/wheel-of-balance-inactive.svg";
+import { IconName } from "~/libs/types/types.js";
+
+const iconNameToSvg: Record<
+	IconName,
+	React.FC<React.SVGProps<SVGSVGElement>>
+> = {
+	myWheelActive: WheelOfBalanceActive,
+	myWheelInactive: WheelOfBalanceInactive,
```
Comment (fshabanov) [inline apps/frontend/src/libs/maps/icon-name-to-svg.map.ts:10]:
```suggestion
	wheelActive: WheelOfBalanceActive,
	wheelInactive: WheelOfBalanceInactive,
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/packages.d.ts (hunk 19 above)
Comment (ihorLysak) [reply apps/frontend/packages.d.ts:36]: If I add this to  vite-env.d.ts, and remove *.svg declaration from packages.d.ts it resolves svg imports to strings

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+		href: AppRoute.ROOT,
+		icon: {
+			active: "myWheelActive",
+			inactive: "myWheelInactive",
+		},
+		title: "My Wheel",
+	},
+];
+
+export { sidebarItems };
```
Comment (JD-Rosales) [inline apps/frontend/src/libs/constants/sidebar-items.constant.ts:15]: this should be `export { SIDEBAR_ITEMS }`

**Review by JD-Rosales — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/constants/sidebar-items.constant.ts (hunk 23 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/constants/sidebar-items.constant.ts:15]: It is exported like that on my local branch, I did a commit on GitHub itself on one of code suggestions, but it didn't go well

File: apps/frontend/packages.d.ts (hunk 19 above)
Comment (fshabanov) [reply apps/frontend/packages.d.ts:36]: According to docs this should be added. Add this and keep the declaration in .d.ts file

File: apps/frontend/src/pages/my-wheel/my-wheel.tsx [hunk 24]
```diff
@@ -0,0 +1,13 @@
+import { Sidebar } from "~/libs/components/components.js";
+
+import styles from "./styles.module.css";
+
+const MyWheel: React.FC = () => {
+	return (
+		<div className={styles["pageContainer"]}>
+			<Sidebar />
```
Comment (JD-Rosales) [inline apps/frontend/src/pages/my-wheel/my-wheel.tsx:8]: I think we should put the sidebar component inside the app component.

**Review by JD-Rosales — CHANGES_REQUESTED** [review]

File: apps/frontend/packages.d.ts [hunk 25]
```diff
@@ -3,27 +3,21 @@ declare module "eslint-plugin-react" {
 
 	const configs: Record<"jsx-runtime" | "recommended", Required<Linter.Config>>;
 
-	export default {
-		configs,
-	};
+	export { configs };
```
Comment (fshabanov) [inline apps/frontend/packages.d.ts:6]: Try to not have unnecessary changes in your PR

File: apps/frontend/src/libs/components/icon/icon.tsx [hunk 26]
```diff
@@ -0,0 +1,14 @@
+import { iconNameToSvg } from "~/libs/maps/maps.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/icon/icon.tsx:1]: Should be in `./libs/maps/maps.js`

File: apps/frontend/src/libs/types/sidebar-item.type.ts [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+
+import { type IconName } from "./icon-name.type.js";
+
+type SidebarItem = {
+	href: ValueOf<typeof AppRoute>;
+	icon: {
+		active: IconName;
+		inactive: IconName;
+	};
+	title: string;
```
Comment (fshabanov) [inline apps/frontend/src/libs/types/sidebar-item.type.ts:12]:
```suggestion
	label: string;
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/packages.d.ts (hunk 25 above)
Comment (ihorLysak) [reply apps/frontend/packages.d.ts:6]: I did this because linter wouldn't let me make a commit

Comment (fshabanov) [reply apps/frontend/packages.d.ts:6]: I understand, but try to find out why did linter start complaining about that? Because previously it was okay, now suddenly it's not

Comment (ihorLysak) [reply apps/frontend/packages.d.ts]: Strange, everything works fine now, but when pushing commit [f8b7660](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/pull/115/commits/f8b766023698543b7c69ff4b487cfe3af84687ec) the CI step for code linting failed with default exports

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 28]
```diff
@@ -0,0 +1,32 @@
...
+import { useLocation } from "~/libs/hooks/hooks.js";
+
+import { SidebarLink } from "./libs/components/components.js";
+import styles from "./styles.module.css";
+
+const Sidebar: React.FC = () => {
+	const { pathname } = useLocation();
+
+	return (
+		<div className={styles["sidebarContainer"]}>
```
Comment (yulyaolshanska) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:11]: I think we should use class name like sidebar-container. Because this naming style is already used in other files

File: apps/frontend/src/libs/components/sidebar/styles.module.css [hunk 29]
```diff
@@ -0,0 +1,16 @@
+.sidebarContainer {
+	background-color: var(--background-gray);
+}
+
+.logoContainer {
+	padding-top: 24px;
```
Comment (yulyaolshanska) [inline apps/frontend/src/libs/components/sidebar/styles.module.css:6]: You can write padding: 24px 0 25px 30px

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 30]
```diff
@@ -0,0 +1,32 @@
...
+				{SIDEBAR_ITEMS.map((item) => {
+					const { active, inactive } = item.icon;
+
+					return (
+						<SidebarLink
+							iconName={item.href === pathname ? active : inactive}
+							key={item.label}
+							label={item.label}
+							pathname={pathname}
+							to={item.href}
```
[the remark this answers is not in the document]
Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/sidebar.tsx:23]: Aren't they already destructured in SidebarLink?

File: apps/frontend/src/libs/components/sidebar/styles.module.css (hunk 29 above)
Comment (ihorLysak) [reply apps/frontend/src/libs/components/sidebar/styles.module.css:6]: it passes the lint stage, but linter has rules for no 0 values. While it passes when 0 provided without px, but I feel like it beats the reason for which this rule is present

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx (hunk 30 above)
[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/frontend/src/libs/components/sidebar/sidebar.tsx]: They are, but they are not destructured in Sidebar component

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the main branch into yours
